tests/AbstractLexerTest.php

```diff
@@ -219,19 +219,12 @@ public function testIsNextToken(string $input, array $expectedTokens): void
      */
     public function testIsNextTokenAny(string $input, array $expectedTokens): void
     {
-        $allTokenTypes = array_map(static function ($token): string {
-            assert($token->type !== null);
-
-            return $token->type;
-        }, $expectedTokens);
-
         $this->concreteLexer->setInput($input);
 
         $this->concreteLexer->moveNext();
         for ($i = 0; $i < count($expectedTokens); $i++) {
             assert($expectedTokens[$i]->type !== null);
             $this->assertTrue($this->concreteLexer->isNextTokenAny([$expectedTokens[$i]->type]));
-            $this->assertTrue($this->concreteLexer->isNextTokenAny($allTokenTypes));
```

Comment:
I don't fully understand how Psalm understands that this is always going to be `true`. The message it gave is: `Docblock-defined type Doctrine\Common\Lexer\Token<string> for $this->concreteLexer->lookahead is never null`
But we never check that `moveNext()` returns `true` here :thinking: 
:bulb: If I remove the line before though, Psalm is happy with keeping this line, so it must have inferred that from the assert on `isNextTokenAny`. So it boils down to: why does Psalm believe `$this->concreteLexer->lookahead->isA(...$allTokenTypes)`  is always going to be true?

Reply:
I'm not entirely convinced we should remove this assertion tbh.